src/pseudopeople/utilities.py

```diff
@@ -19,40 +18,37 @@
     from pseudopeople.schema_entities import DatasetSchema
 
 
-def get_randomness_stream(dataset_name: str, seed: Any, index: pd.Index) -> RandomnessStream:
-    # Avoid error from empty index
-    idx_max = max(index) if not index.empty else 1
-    map_size = max(1_000_000, idx_max * 2)
-    return RandomnessStream(
-        key=dataset_name,
-        clock=lambda: pd.Timestamp("2020-04-01"),
-        seed=seed,
-        index_map=IndexMap(size=map_size),
-    )
+def get_hash(key: str) -> int:
+    max_allowable_numpy_seed = 4294967295  # 2**32 - 1
+    return int(hashlib.sha1(key.encode("utf8")).hexdigest(), 16) % max_allowable_numpy_seed
+
+
+def get_random_generator(dataset_name: str, seed: Any) -> np.random.default_rng:
+
+    key = "_".join([dataset_name, str(seed)])
+    return np.random.default_rng(get_hash(key))
 
 
 def vectorized_choice(
     options: Union[list, pd.Series],
     n_to_choose: int,
-    randomness_stream: RandomnessStream,
+    random_generator: np.random.default_rng,
```

Comment:
nit: consider making `RandomGenerator = np.random.default_rng` for type hinting. Not a big deal either way, but would be less ugly